lib/sarif/sarif_report.rb

```diff
@@ -0,0 +1,42 @@
+require 'json'
+require_relative './base_sarif'
+module Sarif
+  # Class for generating sarif reports
+  class SarifReport
+    SARIF_VERSION = "2.1.0".freeze
+
+    SARIF_SCHEMA = "https://schemastore.azurewebsites.net/schemas"\
+      "/json/sarif-2.1.0-rtm.5.json".freeze
+
+    def initialize(scan_reports)
+      @scan_reports = scan_reports
+    end
+
+    # Builds Sarif Report
+    #
+    # @return [JSON]
+    def to_sarif
+      sarif_report = {
+        "version" => SARIF_VERSION,
+        "$schema" => SARIF_SCHEMA,
+        "runs" => []
+      }
+      # for each scanner report, run the appropriate converter
+      @scan_reports.each { |scan_report| sarif_report["runs"] << converter(scan_report[0]) }
+      JSON.pretty_generate(sarif_report)
+    end
+
+    # Converts a ScanReport to a sarif report for the given scanner
+    #
+    # @params sarif_report [Salus::ScanReport]
+    # @return
+    def converter(scan_report)
+      adapter = "Sarif::#{scan_report.scanner_name.capitalize}Sarif"
+      converter = Object.const_get(adapter).new(scan_report)
+      converter.sarif_report
+    rescue NameError
```

Comment:
If I understand your intention correctly, `NameError` is only supposed to associate with `Object.const_get`.
But why is `converter.sarif_report` also part of the `begin ...` block?

Reply:
The goal was to have the converter.sarif_report returned in either situation (If an adapter exists or if an adapter does not exist). Is there a cleaner way to achieve this? since I am repeating the same thing twice?

Reply:
I would move `converter.sarif_report` out of the the `begin...` block.

You can just return empty sarif report if there is an NameError, like
```ruby
begin ...
rescue NameError
  # return empty sarif report
end

converter.sarif_report
```

Now, a `return` inside `rescue` should work (unless you have an `ensure` block too).  But I'm not 100% sure.  Please test it.

Reply:
I tried an ensure block but it does not return anything. 

The rescue block is run if the adapter for a scanner cannot be instantiated. It would run the sarif_report for the BaseSarif class. However, if the converter/adapter can be instantiated, then it would run the sarif_report for that scanner's adapter (that is it would run GosecSarif.sarif_report instead of BaseSarif.sarif_report). 

I tried your suggestion but I had an issue. Having the function call in the rescue block alone would mean an instantiated adapter would not return a sarif_report. Because of this i call the sarif_report on both blocks(begin and rescue) because its called on two different objects